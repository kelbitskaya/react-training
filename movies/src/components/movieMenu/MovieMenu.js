import React, { useState } from 'react';
import PropTypes from 'prop-types';
import CloseButton from '../closeButton/CloseButton';

export default function MovieMenu(props) {
  const { deleteMovie, editMovie } = props;
  const [isOpen, setIsOpen] = useState(false);

  const showMenu = () => {
    setIsOpen(!isOpen);
  };

  return (
    <>
      {!isOpen
        ? (
          <div className="menu__item" role="presentation" onClick={showMenu}>
            <div className="circle" />
            <div className="circle" />
            <div className="circle" />
          </div>
        )
        : (
          <ul className="dropdown-content dropdown-content__shown">
            <CloseButton handleClose={showMenu} />
            <li role="presentation" className="dropdown-content__item" type="button">
              <button type="button" className="dropdown-content__button" onClick={editMovie}>Edit</button>
            </li>
            <li role="presentation" className="dropdown-content__item" type="button">
              <button type="button" className="dropdown-content__button" onClick={deleteMovie}>delete</button>
            </li>
          </ul>
        )}
    </>
  );
}

MovieMenu.propTypes = {
  editMovie: PropTypes.func.isRequired,
  deleteMovie: PropTypes.func.isRequired,
};
