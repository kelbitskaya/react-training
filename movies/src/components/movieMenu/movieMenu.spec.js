import React from 'react';
import rerender from 'react-test-renderer';
import MovieMenu from './MovieMenu';

const props = {
  editMovie: jest.fn(),
  deleteMovie: jest.fn(),
  isOpen: true
};

describe("should render MovieMenu component", ()=>{
  it("should contain menu__item wrapper", ()=> {
    const component = rerender.create(<MovieMenu {...props}/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});


