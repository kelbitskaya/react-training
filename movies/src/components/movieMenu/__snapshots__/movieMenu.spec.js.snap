// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MovieMenu component should contain menu__item wrapper 1`] = `
<div
  className="menu__item"
  onClick={[Function]}
  role="presentation"
>
  <div
    className="circle"
  />
  <div
    className="circle"
  />
  <div
    className="circle"
  />
</div>
`;
